Modules/_ctypes/callproc.c

```diff
@@ -1123,6 +1123,15 @@ PyObject *_ctypes_callproc(PPROC pProc,
                 goto cleanup; /* leaking ? */
             }
         }
+#if (defined(__x86_64__) && (defined(__MINGW64__) || defined(__CYGWIN__))) || \
+        defined(__aarch64__)
+        void *tmp;
+        if (pa->ffi_type->type == FFI_TYPE_STRUCT) {
```

Comment:
@vsajip Okay, I think you're right here--the issue is basically the same as the old libffi_msvc code that's backported into ctypes--my changes actually cause a segfault if a small (<=8 bytes) struct is passed by value, so I should add a test for that.